### PR TITLE
Bug 1356905 - Fix serialization

### PR DESCRIPTION
**Pull request by rnewman**
Here's the failing test.

### CODE REVIEW

**Comment by farhanpatel** [comment]
### SwiftLint found issues

#### Errors

File | Line | Reason |
| --- | ----- | ----- |
HistorySynchronizerTests.swift | 221 | Opening braces should be preceded by a single space and on the same line as the declaration. 


<p align="right" data-meta="generated_by_danger">
  Generated by :no_entry_sign: <a href="http://danger.systems/">Danger</a>
</p>

**Comment by rnewman** [comment]
@sleroux, @justindarc: you might want to audit all other uses of `.rawString`, including via `.stringValue()`. It apparently does not work correctly.

**Comment by farhanpatel** [comment]
### SwiftLint found issues

#### Errors

File | Line | Reason |
| --- | ----- | ----- |
HistorySynchronizerTests.swift | 221 | Opening braces should be preceded by a single space and on the same line as the declaration. 


<p align="right" data-meta="generated_by_danger">
  Generated by :no_entry_sign: <a href="http://danger.systems/">Danger</a>
</p>

File: Sync/KeyBundle.swift [hunk 1]
```diff
@@ -172,11 +172,29 @@ open class KeyBundle: Hashable {
     open func serializer<T: CleartextPayloadJSON>(_ f: @escaping (T) -> JSON) -> (Record<T>) -> JSON? {
         return { (record: Record<T>) -> JSON? in
             let json = f(record.payload)
-            let data = (json.rawString([writtingOptionsKeys.castNilToNSNull: true]) ?? "").utf8EncodedData
+            if json.isNull() {
```
Comment (sleroux) [inline Sync/KeyBundle.swift]: Would it make sense to move this to an extension on SwiftyJSON so we can use it in other places we `.utf8Encoded` SwiftJSON values? I see we also do this in `FxAClient10.swift` and `PushClient.swift`.

Comment (rnewman) [inline Sync/KeyBundle.swift]: Quite possibly, yes. I wanted to limit the scope of this change, given the limited duration of testing — and FxA (and push?) tend not to encode arbitrary strings from the internet!

I left a note in the PR description or the bug comment suggesting that we look elsewhere, but I'd rather see that done as a follow-up.

**Pull request merged**